src/types.ts

```diff
@@ -90,6 +90,17 @@ export interface SmartweaveDataProvider {
     id: ArweaveTransactionID,
     wallet: ArweaveTransactionID,
   ): Promise<number>;
+  deployContract({
+    srcCodeTransactionId,
+    initialState,
+    stateType, // 'TXID', 'TAG', 'DATA' - configures how to apply the state to the transaction
+    tags,
+  }: {
+    srcCodeTransactionId: ArweaveTransactionID;
+    initialState: ANTContractJSON | ArweaveTransactionID;
+    stateType: 'TXID' | 'TAG' | 'DATA';
```

Comment:
within the ArNS Portal - we'll likely only be using `DATA`  -  do we need to support these other methods? i know they are available in smartweave but out of scope for how we'll deploy contracts